### ISSUE TITLE
Two suggestions for sdmxdf

### ISSUE THREAD

**Issue by Tungurahua** [amattioc/SDMX#59]
The `sdmxdf` function returns SDMX-data in a dataframe. A session looks like this:

    > result = getTimeSeries('EUROSTAT', 't2020_10/A.T.')
    Apr 15, 2015 7:59:17 PM it.bancaditalia.oss.sdmx.client.RestSdmxClient runQuery
    INFO: Contacting web service with query: http://ec.europa.eu/eurostat/SDMX/diss ...

    > head(sdmxdf(result,meta = T))
                   ID TIME  OBS STATUS FREQ SEX GEO
    1 t2020_10.A.T.AT 1992  NaN     na    A   T  AT
    2 t2020_10.A.T.AT 1993  NaN     na    A   T  AT
    3 t2020_10.A.T.AT 1994 70.6           A   T  AT
    4 t2020_10.A.T.AT 1995 71.2           A   T  AT

If `meta = T`, the ID column contains redundant information (except for the data code) and will probably be discarded in most cases. This could be improved by adding a further argument `id=F` to the function that  strips the id from the results by default and allows adding it if necessary. I was thinking about something like this:

    sdmxdf <- function (tslist, meta = F, id = F) 
    {
      ddf = NULL
      if (!missing(tslist) && length(tslist) != 0 && is.list(tslist)) {
        dflist = lapply(tslist, RJSDMX:::sdmxzoo2df, meta)
        ddf = Reduce(function(x, y) merge(x, y, all = TRUE), 
                     dflist)
        rownames(ddf) <- NULL
      if(id==F){
        ddf  <-  ddf[ , -which(names(ddf) %in% c("ID"))]
      }
    [...] 

Though `sdmxzoo2df()` would probably be the right place to implement this.

A similar argument could apply to SDMX attributes like the  `STATUS` field , however this would only make sense if this field is defined as a standard-field with a fixed name in SDMX. The argument could be used to define if only dimensions are retrieved or also attributes.

Another suggestion concerns the order in which the variables are returned from `sdmxdf()`. I think it would be 'more standard' to have the factors to the left and the `OBS` column as the rightmost column (with `GEO` and `TIME` left of it). While this is probably mostly a matter of taste, I think that some functions e.g. in the `plyr` package expect the value column to be the rightmost column by default.

**Comment by amattioc** [comment]
Hi, 

actually the id is not completely redundant because in the current implementation there is no difference between dimension and attribute columns. This means that there is no way to understand the id of the time series in the df form.

Regarding the attributes, you are right that the STATUS is treated as a known object, but this is an error ( #28)  and will be fixed soon (more or less :-)) In SDMX all metadata have to be dynamically discovered, as they depend only on the DSD of the data flow (e.g GEO is not always present). For this same reason the order of the dimensions is important as it intrinsically defines the time series identifier.

So, starting from your suggestions I could:

* add 3 new parameters (*id*, *dimensions*, *attributes*) and remove *metadata* to selectively enable metadata
* add an R attribute (*type*) to every metadata column (values: *attribute* or *dimension*). This will make it possible to state the nature of metadata and build the id
* put time and value at the end of the df.

What about that?

Best, Attilio

**Comment by Tungurahua** [comment]
Hi Attilio,
thanks again for the quick reply and the detailed explanations. Adding variable-level attributes to state the type (attribute or dimension) is a good idea. In addition the name of the dataset (aka dataflow) could be set as an attribute of the entire dataframe. In my opinion this would improve the readability of the retrieved data. 

Best
Albrecht

**Comment by amattioc** [comment]
Hi, it makes sense. Thanks

**Comment by amattioc** [comment]
Hi @Tungurahua I just pushed commit 54e0642ca6234960ded3f97a940c299761443850 that contains some changes related to this issue. I decided to keep it simple, following your initial suggestion. Now you have the `id` parameter, that defaults to TRUE and allows you to remove the related column. The `meta` parameter works as usual.

I could not add the *dataflow id* to the data.frame as the `getTimeSeries` function allows to query object from multiple dataflows, e.g.:

`result = getSDMX('ECB', 'EXR.A.USD.EUR.SP00.A;MOBILE_EXR.D.USD.EUR.SP00.A')`

Let me know if the change is ok.

Best, Attilio
